### ISSUE TITLE
is batch pad ignored in evaluation???

### ISSUE THREAD

**Issue by Pagey** [apache/incubator-mxnet#6769]
i am working with a small validation set, which is not far from my batch size
trying to understand the validation metric i'm getting;

i'm working with mx.io.NDArrayIter with default last_batch_handle, which is 'pad'

i can't find in the code a place which calculates the update for the eval_data metrics which refer to this pad- is there a chance that it is ignored, and the last batch on the validation data also contains the "rolled over" samples from the beginning of the validation set??? this may bias results for those working with small datasets

a fix could change the size of the last batch going into the update so it doesn't include the pad?
************************************
[edit- looks like  iter_predict does refer to this (in base_module.py)

`        for nbatch, eval_batch in enumerate(eval_data):
            if num_batch is not None and nbatch == num_batch:
                break
            self.forward(eval_batch, is_train=False)
            pad = eval_batch.pad
            outputs = [out[0:out.shape[0]-pad] for out in self.get_outputs()]
`
     
...

but in score:
`        for nbatch, eval_batch in enumerate(eval_data):
            if num_batch is not None and nbatch == num_batch:
                break

            self.forward(eval_batch, is_train=False)
            self.update_metric(eval_metric, eval_batch.label)
`
 
...
************************************

PS this could be solved if i could change the validation batch size to be the entrire validation set, but this doesn't seem to work- is there still a problem with different training and validation batch sizes? as mentioned here [https://github.com/dmlc/mxnet/issues/648](url)
??
i didn't get an error but the eval calucaltion seemed to work with the training batch size??

thanks,

version: mxnet 0.10.1

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
